pyvista/plotting/lights.py

```diff
@@ -0,0 +1,465 @@
+"""Module containing pyvista implementation of vtkLight."""
+
+from enum import IntEnum
+
+import vtk
+from vtk import vtkLight
+
+import pyvista
+
+class LightType(IntEnum):
+    """An enumeration for the light types."""
+
+    HEADLIGHT = 1
+    CAMERA_LIGHT = 2
+    SCENE_LIGHT = 3
+
+    def __str__(self):
+        """Pretty name for a light type."""
+        return self.name.replace('_', ' ').title()
+
+class Light(vtkLight):
+    """Light class.
+
+    Parameters
+    ----------
+    position : list or tuple, optional
+        The position of the light. The interpretation of the position depends
+        on the type of the light.
+
+    color : list or tuple, optional
+        The color of the light. The ambient, diffuse and specular colors will
+        all be set to this color in creation.
+
+    light_type : string or int, optional
+        The type of the light. If a string, one of ``'headlight'``,
+        ``'camera light'`` or ``'scene light'``. If an int, one of 1, 2 or 3,
+        respectively. The class constants ``Light.HEADLIGHT``, ``Light.CAMERA_LIGHT``
+        and ``Light.SCENE_LIGHT`` are also available, respectively.
+
+        A headlight is attached to the camera, looking at its focal point along
+        the axis of the camera.
+        A camera light also moves with the camera, but it can occupy a general
+        position with respect to it.
+        A scene light is stationary with respect to the scene, as it does not
+        follow the camera. This is the default.
+    """
+
+    # TODO: better/more explanation for ``position``?
+
+    # pull in light type enum values as class constants
+    HEADLIGHT = LightType.HEADLIGHT
+    CAMERA_LIGHT = LightType.CAMERA_LIGHT
+    SCENE_LIGHT = LightType.SCENE_LIGHT
+    # TODO: is this design OK?
```

Comment:
I would just use the enum directly.

Reply:
The reason I wanted to pull these in into the class is because I can easily imagine use cases where we want to compare light types from outside the `Light` class. Say, if I end up with a list of light types for some reason. In this case I'd want to be able to do `pyvista.Light.HEADLIGHT`. I guess we could also pull in `LightType` into the pyvista namespace, but to me that would seem overkill (it's not important enough to pollute the main namespace with it).